pkg/virt-api/rest/subresource_test.go

```diff
@@ -2271,31 +2279,49 @@ var _ = Describe("VirtualMachineInstance Subresources", func() {
 			app.PauseVMIRequestHandler(request, response)
 
 			Expect(response.StatusCode()).To(Equal(http.StatusOK))
-
+			//In case of dry-run the request is not propagated to the handler
+			Expect(backend.ReceivedRequests()).To(handlerExpectation)
 		},
-			Entry("with default", &v1.PauseOptions{}),
-			Entry("with dry-run option", &v1.PauseOptions{DryRun: getDryRunOption()}),
+			Entry("with default", &v1.PauseOptions{}, HaveLen(1)),
+			Entry("with dry-run option", &v1.PauseOptions{DryRun: getDryRunOption()}, BeNil()),
 		)
 
-		DescribeTable("Should fail pausing", func(running bool, paused bool, pauseOptions *v1.PauseOptions) {
+		withLivenessProbe := func(vmi *v1.VirtualMachineInstance) {
+			vmi.Spec.LivenessProbe = &v1.Probe{
+				Handler:             v1.Handler{},
+				InitialDelaySeconds: 120,
+				TimeoutSeconds:      120,
+				PeriodSeconds:       120,
+				SuccessThreshold:    1,
+				FailureThreshold:    1,
+			}
+		}
+		nilAdditionalOps := func(vmi *v1.VirtualMachineInstance) {
+			return
+		}
 
-			expectVMI(running, paused)
+		DescribeTable("Should fail pausing", func(running bool, paused bool, additionalOpts func(vmi *v1.VirtualMachineInstance), pauseOptions *v1.PauseOptions, expectedCode int, expectedError string) {
+			expectVMI(running, paused, additionalOpts)
```

Comment:
If you make additionalOpts variadic you should be able to get away without adding `nilAdditionalOps`? The third arg on `expectVMI` is already variadic.

Reply:
Yes, but I like to have parameters related to the same context close together.
Actually, probably there are too many params here, and this could be improved, but I would leave it for a follow-up. WDYT?

Reply:
Sounds like a plan.